inception-active-learning/src/main/java/de/tudarmstadt/ukp/inception/active/learning/ActiveLearningServiceImpl.java

```diff
@@ -102,4 +112,266 @@ public ActiveLearningServiceImpl(DocumentService aDocumentService,
         return model.getFlattenedPredictions(aState.getDocument().getName(), aSelectedLayer,
             windowBegin, windowEnd, aJcas, true);
     }
+
+    @Override
+    public void setSessionActive(User aUser, boolean aSesscionActive)
```

Comment:
If you make `getState(user)` public and allow then you could wrap `ActiveLearningUserState` it in a CompoundPropertyModel and use it in the forms. All the getters and setters here would not be needed and also `setUserState` would go away.

Reply:
Thanks @reckart I'm still not sure whether I understand it correctly. 

For example, in ActiveLearningSidebar, I created a field `private CompoundPropertyModel<ActiveLearningUserState> userStateModel;` and assigned this field `userStateModel = new CompoundPropertyModel<>(activeLearningService.getState(user.getUsername()));` when active learning sidebar generated.

ActiveLearningSiebar has a `sessionControlForm`, it contains a DropDownChoice to select the `AnnotationLayer` and a LambdaAjaxButton to show terminate/start and change the field `sessionActive`. (AnnotationLayer selectedLayer and boolean sessionActive are fields in ActiveLearningUserState.)

For the DropDownChoice, I think I should write:
```
DropDownChoice<AnnotationLayer> layersDropdown = new DropDownChoice<AnnotationLayer>(CID_SELECT_LAYER, userStateModel.bind("selectedLayer"));
layersDropdown.setChoices(LambdaModel.of(this::listLayersWithRecommenders));
```
This component is disabled when `sessionActive` is true. How can I bind `sessionActive` also to the DropDownChoice? Using `userStateModel.getObject().isSessionActive`?

For the second component LambdaAjaxButton, I wrote like this:
 ```
LambdaAjaxButton<Void> startStopButton = new LambdaAjaxButton<>(
                CID_LAYER_SELECTION_BUTTON, this::actionStartStopTraining);
startStopButton.setModel(userStateModel.bind("sessionActive"));
```
How could I change the button value according to "sessionActive"? (Before:         `startStopButton.setModel(LambdaModel.of(() -> sessionActive ? "Terminate" : "Start"));`)
How could I set `sessionActive` in the method `actionStartStopTraining`?

Reply:
> For example, in ActiveLearningSidebar, I created a field private CompoundPropertyModel<ActiveLearningUserState> userStateModel; and assigned this field userStateModel = new CompoundPropertyModel<>(activeLearningService.getState(user.getUsername())); when active learning sidebar generated.

Instead of fetching the state there, I would wrap the state fetching into a `LambdaModelAdapter`, i.e. something like `new CompoundPropertyModel<>(LambdaModelAdapter.of(() -> activeLearningService.getState(user.getUsername()), state -> activeLearningService.setState(user.getUsername(), state))`. That directly binds the model to the underlying data from the ActiveLearningService.

In order to activate/deactivate a session, I think you can fetch the active learning state from the model and update it directly, e.g. `userStateModel.getModelObject().setActive(true/false).

In order to change the label of the `startStopButton`, I would try using a `LambdaBehavior`, e.g. `startStopButton.add(LambdaBehavior.configure(_component -> setModelObject(userStateModel.getModelObject().iActive() ? "Terminate" : "Start"));`

Does that help?